src/jsonformat.erl

```diff
@@ -100,12 +100,12 @@ apply_format_funs(Data, _) ->
   Data.
 
 apply_key_mapping(Data, #{key_mapping := Mapping}) ->
-  DataOnlyMapped = 
-    maps:fold(fun(K, V, Acc) when is_map_key(K, Data) -> Acc#{V => maps:get(K, Data)};
-                 (_, _, Acc)                          -> Acc
-              end, #{}, Mapping),
-  DataNoMapped = maps:without(maps:keys(Mapping), Data),
-  maps:merge(DataNoMapped, DataOnlyMapped);
+  maps:fold(fun(Key, NewKey, Acc0) ->
+                case maps:take(Key, Acc0) of
+                  {Val, Acc} -> Acc#{NewKey => Val};
+                  error -> Acc0
+                end
+            end, Data, Mapping);
```

Comment:
can this handle this case?

```erlang
#{key_mapping => #{ level => text
                              , text => level}},
```

Reply:
If we have a log `{"level": "1", "text": "2" }`, what is the expected output? Is is `{"text": "1", "level": "2"}`?

Reply:
I'd say so. but that's more of an opinion than an obvious fact I guess

Reply:
I think that behaviour makes sense, and this use case cannot be handled by my implementation above. You're right.

Reply:
Let's keep the current implementation, I'll close this PR.